### PR TITLE
Zth  newvendingmachines

### PR DESCRIPTION
**Pull request by d3athrow**
Custom vending machines, fixed the issue with setting the original account number.

### CODE REVIEW

**Comment by ghost** [comment]
Cool

**Comment by Knightkiller11** [comment]
Jesus whats with all the commits? Half of them don't seem related to custom vending machines at all.

**Comment by SonixApache** [comment]
dem commits tho

**Comment by SkowronX** [comment]
wew

**Comment by d3athrow** [comment]
Yeah I know gonna fix it when I have a chance

**Comment by d3athrow** [comment]
K nice n clean

**Comment by gbasood** [comment]
Needs an actual descriptive changelog

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -502,6 +502,19 @@ Class Procs:
 	update_icon()
 	return 1
 
+/obj/machinery/proc/toggleSecuredPanelOpen(var/obj/toggleitem, var/mob/user)
+	if(!linked_account || panel_open)
+		togglePanelOpen(toggleitem, user)
+		return 1
+	var/account_try = input(user,"Please enter the already connected account number to unlock the panel","Security measure") as null|num
+	if(!user.Adjacent(src))
```
Comment (N3X15) [inline code/game/machinery/machinery.dm]: Should check adjacency prior to input(), as well.

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -423,23 +448,31 @@ var/global/num_vending_terminals = 1
...
-			else
-				to_chat(usr, "[bicon(src)]<span class='warning'>Unable to connect to linked account.</span>")
+		if(linked_account)
+			if(account_first_linked)
+				var/account_try = input(user,"Please enter the already connected account number","Security measure") as num
+				if(account_try != linked_account.account_number)
+					to_chat(user, "[bicon(src)]<span class='warning'>Access denied. Your input doesn't match the vending machine's connected account.</span>")
+					return
+			var/new_account = input(user,"Please enter the account to connect to.","New account link") as num
+			for(var/datum/money_account/D in all_money_accounts)
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: Adjacency check before and after input().

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -423,23 +448,31 @@ var/global/num_vending_terminals = 1
...
-			reconnect_database()
-		if(linked_db)
-			if(linked_account)
-				var/obj/item/weapon/card/I = W
-				scan_card(I)
-			else
-				to_chat(usr, "[bicon(src)]<span class='warning'>Unable to connect to linked account.</span>")
+		if(linked_account)
+			if(account_first_linked)
+				var/account_try = input(user,"Please enter the already connected account number","Security measure") as num
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: Adjacency check before and after input().

File: code/game/objects/items/stacks/stack_recipes.dm [hunk 4]
```diff
@@ -228,21 +228,22 @@ var/list/datum/stack_recipe/wood_recipes = list (
...
-	new/datum/stack_recipe("crayon box",		/obj/item/weapon/storage/fancy/crayons/empty			),
-	new/datum/stack_recipe("cardborg suit",		/obj/item/clothing/suit/cardborg,					3	),
-	new/datum/stack_recipe("cardborg helmet",	/obj/item/clothing/head/cardborg						),
-	new/datum/stack_recipe("pizza box",			/obj/item/pizzabox										),
-	new/datum/stack_recipe("folder",			/obj/item/weapon/folder									),
-	new/datum/stack_recipe("flare box",			/obj/item/weapon/storage/fancy/flares/empty				),
-	new/datum/stack_recipe("donut box",			/obj/item/weapon/storage/fancy/donut_box/empty			),
-	new/datum/stack_recipe("eggbox",			/obj/item/weapon/storage/fancy/egg_box/empty			),
-	new/datum/stack_recipe("paper bin",			/obj/item/weapon/paper_bin/empty						),
+	new/datum/stack_recipe("box",							/obj/item/weapon/storage/box							),
```
Comment (N3X15) [inline code/game/objects/items/stacks/stack_recipes.dm]: Hard tabs are cancer, if you align stuff, use spaces.

**Review by N3X15 — CHANGES_REQUESTED** [review]
Add the adjacency checks and you should be golden.

**Review by N3X15 — APPROVED** [review]

**Comment by Zth--** [comment]
cool

**Pull request merged**